id: PYSEC-0000-CVE-2022-23580
details: Tensorflow is an Open Source Machine Learning Framework. During shape inference,
  TensorFlow can allocate a large vector based on a value from a tensor controlled
  by the user. The fix will be included in TensorFlow 2.8.0. We will also cherrypick
  this commit on TensorFlow 2.7.1, TensorFlow 2.6.3, and TensorFlow 2.5.3, as these
  are also affected and still in supported range.
affected:
- package:
    name: tensorflow-cpu
    ecosystem: PyPI
    purl: pkg:pypi/tensorflow-cpu
  ranges:
  - type: GIT
    repo: https://github.com/tensorflow/tensorflow
    events:
    - introduced: "0"
    - fixed: 1361fb7e29449629e1df94d44e0427ebec8c83c7
  - type: ECOSYSTEM
    events:
    - introduced: "0"
    - fixed: 2.5.3
    - introduced: 2.6.0
    - fixed: 2.6.3
references:
- type: FIX
  url: https://github.com/tensorflow/tensorflow/commit/1361fb7e29449629e1df94d44e0427ebec8c83c7
- type: ADVISORY
  url: https://github.com/tensorflow/tensorflow/security/advisories/GHSA-627q-g293-49q7
- type: WEB
  url: https://github.com/tensorflow/tensorflow/blob/a1320ec1eac186da1d03f033109191f715b2b130/tensorflow/core/framework/shape_inference.cc#L788-L790
aliases:
- CVE-2022-23580
- GHSA-627q-g293-49q7
modified: "2022-02-10T01:30:00Z"
published: "2022-02-04T23:15:00Z"
